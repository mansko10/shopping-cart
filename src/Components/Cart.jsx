export default function Cart() {
  return (
    <>
      <h1>This is cart</h1>
    </>
  );
}
