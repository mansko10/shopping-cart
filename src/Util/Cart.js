function addToCart(product, cart, setCart) {
  const includesProduct = cart.some(
    (cartProduct) => cartProduct.id === product.id,
  );

  if (!includesProduct) {
    setCart([...cart, product]);
  } else if (includesProduct) {
    const newCart = cart.map((cartProduct) => {
      if (cartProduct.id === product.id) {
        return {
          ...cartProduct,
          quantity: cartProduct.quantity + product.quantity,
          totalPrice: cartProduct.totalPrice + product.totalPrice,
        };
      } else {
        return cartProduct;
      }
    });

    setCart(newCart);
  }
}

function getCartQuantity(cart) {
  let result = 0;

  cart.forEach((product) => {
    result += product.quantity;
  });

  return result;
}

export { addToCart, getCartQuantity };
