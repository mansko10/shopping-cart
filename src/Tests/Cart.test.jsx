import { describe, vi, expect, it } from "vitest";
import Cart from "../Components/Cart";
import { render, screen } from "@testing-library/react";
import { RenderContext } from "./RenderContext";

describe("Renders Cart correctly", () => {
  it("Renders correctly when cart includes items", () => {
    const dummy = {
      cart: [
        {
          title: "Mens Cotton Jacket",
          quantity: 2,
          price: 55.99,
          totalPrice: 111.98,
          id: 3,
          image: "https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg",
        },
      ],
      updateProductQuantity: vi.fn(),
    };

    const { container } = render(
      <RenderContext context={dummy}>
        <Cart />
      </RenderContext>,
    );

    expect(screen.getByText("Mens Cotton Jacket")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("Renders correctly when cart is empty", () => {
    const dummy = {
      cart: [],
      updateProductQuantity: vi.fn(),
    };

    const { container } = render(
      <RenderContext context={dummy}>
        <Cart />
      </RenderContext>,
    );

    expect(screen.getByText("Cart is Empty")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
