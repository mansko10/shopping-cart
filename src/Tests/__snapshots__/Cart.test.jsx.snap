// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders Cart correctly > Renders correctly when cart includes items 1`] = `
<div>
  <div
    class="sc-blHHSb bopiyA"
  >
    <h1>
      Cart:
    </h1>
    <button
      class="clear-cart"
      type="button"
    >
      Clear Cart
    </button>
    <div
      class="cart"
    >
      <div
        class="cart-item"
      >
        <img
          alt=""
          src="https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"
        />
        <p
          class="title"
        >
          Mens Cotton Jacket
        </p>
        <p
          class="totalPrice"
        >
          $
          111.98
        </p>
        <div
          class="modify-quantity"
        >
          <button
            class="modify"
            type="button"
          >
            -
          </button>
          <input
            min="1"
            type="number"
            value="2"
          />
          <button
            class="modify"
            type="button"
          >
            +
          </button>
        </div>
        <button
          class="delete"
          type="button"
        >
          Remove
        </button>
      </div>
      <p
        class="combined-total"
      >
        Total: $
        111.98
      </p>
      <button
        class="checkout"
        type="button"
      >
        Proceed to Checkout
      </button>
    </div>
  </div>
</div>
`;

exports[`Renders Cart correctly > Renders correctly when cart is empty 1`] = `
<div>
  <div
    class="sc-blHHSb bopiyA"
  >
    <h1>
      Cart:
    </h1>
    <button
      class="clear-cart"
      type="button"
    >
      Clear Cart
    </button>
    <div
      class="cart"
    >
      <p
        style="text-align: center;"
      >
        Cart is Empty
      </p>
    </div>
  </div>
</div>
`;

exports[`Test Cart > abc 1`] = `
<div>
  <div
    class="sc-blHHSb bopiyA"
  >
    <h1>
      Cart:
    </h1>
    <button
      class="clear-cart"
      type="button"
    >
      Clear Cart
    </button>
    <div
      class="cart"
    >
      <div
        class="cart-item"
      >
        <img
          alt=""
          src="https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"
        />
        <p
          class="title"
        >
          Mens Cotton Jacket
        </p>
        <p
          class="totalPrice"
        >
          $
          111.98
        </p>
        <div
          class="modify-quantity"
        >
          <button
            class="modify"
            type="button"
          >
            -
          </button>
          <input
            min="1"
            type="number"
            value="2"
          />
          <button
            class="modify"
            type="button"
          >
            +
          </button>
        </div>
        <button
          class="delete"
          type="button"
        >
          Remove
        </button>
      </div>
      <p
        class="combined-total"
      >
        Total: $
        111.98
      </p>
      <button
        class="checkout"
        type="button"
      >
        Proceed to Checkout
      </button>
    </div>
  </div>
</div>
`;
